smart_contracts/contract/src/no_std_handlers.rs

```diff
@@ -1,25 +1,20 @@
-//! Contains definitions for panic and allocation error handlers, along with other `no_std` support
-//! code.
-#[cfg(feature = "test-support")]
-use crate::contract_api::runtime;
-#[cfg(feature = "test-support")]
-use alloc::format;
+//! Contains definitions for panic and allocation error handlers.
 
 /// A panic handler for use in a `no_std` environment which simply aborts the process.
 #[panic_handler]
 #[no_mangle]
-pub fn panic(_info: &::core::panic::PanicInfo) -> ! {
+pub fn panic(_info: &core::panic::PanicInfo) -> ! {
     #[cfg(feature = "test-support")]
-    runtime::print(&format!("Panic: {}", _info));
-    ::core::intrinsics::abort();
+    crate::contract_api::runtime::print(&alloc::format!("Panic: {}", _info));
```

Comment:
As we discussed offline - with an improved Makefile `build-contracts-rs` we have to remove this line as we've observed negative performance hit on transaction benchmarks by 20% on average